### ISSUE TITLE
firefox-nightly-bin wrapper needs to include libEGL on LD_LIBRARY_PATH

### ISSUE THREAD

**Issue by colemickens** [mozilla/nixpkgs-mozilla#137]
I think that the `firefox-nightly-bin` wrapper `firefox` needs to have LD_LIBRARY_PATH further augmented to include `libglvnd`, for example, to provide the `libEGL`.

Note, it still doesn't work for me, but it seems like it's necessary at least.

**Comment by garbas** [comment]
I think this is a known problem with wayland did a quick search and i came up with this 2 links:
- https://bugzilla.mozilla.org/show_bug.cgi?id=1489902
- https://github.com/xhorak/firefox-devedition-flatpak/issues/90

I think if you disable webrender you would be fine.

I suggest you subscribe to those tickets, and follow progress there.

**Comment by colemickens** [comment]
@garbas Yeah, things are fine with WebRender disabled, and even with libGDL on LD_LIBRARY_PATH, it doesn't seem to work due to the bug you linked on BugZilla.

I made this change in my nixpkgs PR adding a Chromium Wayland build, it should be an easy thing to add to the wrapper here. I will try my hand at it whenever the Firefox WebRender builds are generally usable w/ Wayland.

**Comment by nbp** [comment]
I have been told that wayland support can be enabled by setting the environment variable `MOZ_ENABLE_WAYLAND` which will default to either x11 or wayland based on the commit message which introduce it.

https://hg.mozilla.org/mozilla-central/rev/b9424178ab0b

We should fix that in nixpkgs, and set it to `1` in the Firefox wrapper of Nixpkgs.

**Comment by colemickens** [comment]
I think it's far too early to be doing that. As much as I want to live in a fully-Wayland world, Firefox still has too many issues to default to Wayland (crashes, performance, broken/missing features, etc).

OTOH, adding libEGL  to firefox-nightly can be a benefit in case the user opts into Wayland, and also this will simply be needed for WebRender on NixOS as I understand it...
